src/memory/client/MemoryBrowser.tsx

```diff
@@ -77,21 +77,50 @@ interface State {
   bytesPerRow: number;
   bytesPerGroup: number;
   endianness: 'le' | 'be';
+  childrenNames: { id: number; name: string }[];
 }
 
 export class MemoryBrowser extends React.Component<Props, State> {
   private addressReq = '';
   private lengthReq = '512';
+  private childReq = 0;
+  public static firstTime: boolean = true;
 
   constructor(props: Props) {
     super(props);
     this.state = {
       bytesPerRow: 32,
       bytesPerGroup: 8,
       endianness: 'le',
+      childrenNames: [],
     };
   }
 
+  async getAvailableChilds() {
+    try {
+      const result = await messageBroker.sendGetChildrenNames({
+        command: 'getChildDapNames',
+        args: {
+          child: this.childReq,
+        },
+      });
+      let childArray: {
+        id: number;
+        name: string;
+      }[] = [];
+      result.result?.child?.map(function (val: string, index: number) {
+        childArray.push({ id: index, name: val });
+      });
+      this.setState({
+        childrenNames: childArray,
+      });
```

Comment:
This isn't how to use map, map should transform one thing to another, e.g.:

Please see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map for a comprehensive overview of map in javascript.

```suggestion
      const childrenNames = result.result?.child?.map((val, index) => {id: index, name: val});
      this.setState({
        childrenNames,
      });
```

(you may want `|| []` at the end of the first line if you don't want childrenNames to be undefined.

Reply:
I have updated it.